mpqp/execution/providers/google.py

```diff
@@ -0,0 +1,353 @@
+        sampler=simulator,
+        device=device,
+        calibrations={calibration.timestamp // 1000: calibration},
+    )
+    simulator = SimulatedLocalEngine([sim_processor])
+
+    job_cirq_circuit = job.circuit.to_other_language(
+        Language.CIRQ, processor_id=job.device.value
+    )
+    assert isinstance(job_cirq_circuit, Cirq_circuit)
+
+    if job.job_type == JobType.STATE_VECTOR:
+        raise NotImplementedError(
+            f"Does not handle {job.job_type} for processor for the moment"
+        )
+    elif job.job_type == JobType.OBSERVABLE:
+        raise NotImplementedError(
+            f"Does not handle {job.job_type} for processor for the moment"
+        )
+    elif job.job_type == JobType.SAMPLE:
+        assert isinstance(job.measure, BasisMeasure)
+        if isinstance(job.measure.basis, ComputationalBasis):
+            result_sim = simulator.get_sampler(job.device.value).run(
+                job_cirq_circuit, repetitions=job.measure.shots
+            )
+        else:
+            raise NotImplementedError(
+                "Does not handle other basis than the ComputationalBasis for the moment"
+            )
+    else:
+        raise ValueError(f"Job type {job.job_type} not handled")
+
+    return extract_result(result_sim, job, GOOGLEDevice.CIRQ_LOCAL_SIMULATOR)
+
+
+def extract_result(
+    result: (
+        StateVectorTrialResult
+        | cirq_result
+        | list[float]
+        | list[ObservableMeasuredResult]
+    ),
+    job: Optional[Job] = None,
+    device: Optional[GOOGLEDevice] = None,
+) -> Result:
+    """
+    Extracts the result from the simulation and formats it into an mpqp Result object.
+
+    Args:
+        result : The result of the simulation.
+        job : The original job. Defaults to None.
+        device : The device used for the simulation. Defaults to None.
+
+    Returns:
+        Result: The formatted result.
+
+    Raises:
+        NotImplementedError: If the job is None or the type is not supported.
+        ValueError: If the result type does not match the expected type for the job type.
+    """
+    if job is None:
+        raise NotImplementedError("result from job None is not implemented")
+    else:
+        if job.job_type == JobType.SAMPLE:
+            if not isinstance(result, cirq_result):
+                raise ValueError(
+                    f"result: {type(result)}, must be a cirq_result for job type {job.job_type}"
+                )
+            return extract_result_SAMPLE(result, job, device)
+        elif job.job_type == JobType.STATE_VECTOR:
+            if not isinstance(result, StateVectorTrialResult):
+                raise ValueError(
+                    f"result: {type(result)}, must be a cirq_result for job type {job.job_type}"
+                )
+            return extract_result_STATE_VECTOR(result, job, device)
+        elif job.job_type == JobType.OBSERVABLE:
+            if isinstance(result, cirq_result):
```

Comment:
Are you checking the right type ? In the Value error you say list[float] or list[Observable..] but here you put cirq_result